internal/gatewayapi/translator.go

```diff
@@ -657,6 +660,160 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
+										)
+										continue
+									}
+
+									// Check if the header to be set has already been configured
+									headerKey := string(setHeader.Name)
+									canAddHeader := true
+									for _, h := range addRequestHeaders {
+										if strings.EqualFold(h.Name, headerKey) {
+											canAddHeader = false
+											break
+										}
+									}
+									if !canAddHeader {
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											fmt.Sprintf("RequestHeaderModifier Filter already configures request header: %s to be added/set, ignoring second entry", headerKey),
+										)
+										continue
+									}
+									newHeader := ir.AddHeader{
+										Name:   string(setHeader.Name),
+										Append: false,
+										Value:  setHeader.Value,
+									}
+
+									addRequestHeaders = append(addRequestHeaders, newHeader)
+								}
+							}
+
+							// Remove request headers
+							// As far as Envoy is concerned, it is ok to configure a header to be added/set and also in the list of
+							// headers to remove. It will remove the original header if present and then add/set the header after.
+							if headersToRemove := headerModifier.Remove; headersToRemove != nil {
+								for _, removedHeader := range headersToRemove {
+									if removedHeader == "" {
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											"RequestHeaderModifier Filter cannot remove a header with an empty name",
+										)
+										continue
+									}
+
+									canRemHeader := true
+									for _, h := range removeRequestHeaders {
+										if strings.EqualFold(h, removedHeader) {
+											canRemHeader = false
+											break
+										}
+									}
+									if !canRemHeader {
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											fmt.Sprintf("RequestHeaderModifier Filter already configures request header: %s to be removed, ignoring second entry", removedHeader),
+										)
+										continue
+									}
+
+									removeRequestHeaders = append(removeRequestHeaders, removedHeader)
+
+								}
+							}
+
+							// Update the status if the filter failed to configure any valid headers to add/remove
+							if len(addRequestHeaders) == 0 && len(removeRequestHeaders) == 0 {
+								parentRef.SetCondition(
+									v1beta1.RouteConditionResolvedRefs,
+									metav1.ConditionFalse,
+									v1beta1.RouteReasonUnsupportedValue,
+									"RequestHeaderModifier Filter did not provide valid configuration to add/set/remove any headers",
+								)
+							}
+						}
```

Comment:
This is inside of the `if headerModifier != nil {` block, which means it won't error for `requestHeaderModifier: null` but will error for `requestHeaderModifier: {}`, which just seems wrong to me.

My reading of the Gateway API spec is that an empty config is valid, and so we shouldn't error on it. (but maybe I missed something?)